crates/web5/src/dids/methods/did_jwk.rs

```diff
@@ -1,117 +1,246 @@
-use super::{MethodError, Result};
 use crate::{
-    crypto::jwk::Jwk,
+    crypto::{
+        dsa::{ed25519::Ed25519Generator, secp256k1::Secp256k1Generator, Dsa},
+        jwk::Jwk,
+        key_managers::{in_memory_key_manager::InMemoryKeyManager, KeyManager},
+    },
     dids::{
+        bearer_did::BearerDid,
         data_model::{document::Document, verification_method::VerificationMethod},
         did::Did,
         resolution::{
-            resolution_metadata::{ResolutionMetadata, ResolutionMetadataError},
-            resolution_result::ResolutionResult,
+            resolution_metadata::ResolutionMetadataError, resolution_result::ResolutionResult,
         },
     },
+    errors::Result,
 };
 use base64::{engine::general_purpose, Engine as _};
+use std::sync::Arc;
 
-#[derive(Clone)]
-pub struct DidJwk {
-    pub did: Did,
-    pub document: Document,
+#[derive(Default)]
+pub struct DidJwkCreateOptions {
+    pub key_manager: Option<Arc<dyn KeyManager>>,
+    pub dsa: Option<Dsa>,
 }
 
+pub struct DidJwk;
+
 impl DidJwk {
-    pub fn from_public_jwk(public_jwk: Jwk) -> Result<Self> {
+    pub fn create(options: Option<DidJwkCreateOptions>) -> Result<BearerDid> {
+        let options = options.unwrap_or_default();
+
+        let key_manager = match options.key_manager {
+            Some(km) => km,
+            None => Arc::new(InMemoryKeyManager::new()),
+        };
+
+        let private_jwk = match options.dsa {
```

Comment:
doing `match options.dsa.unwrap_or(Dsa::Ed25519)` might make this cleaner, but either one works

Reply:
done